client/components/tag/index.js

```diff
@@ -3,63 +3,104 @@
 import './style.scss';
 
 /**
- * This component can be used to show an item styled as a "tag", optionally with an `X` + "remove".
- * Generally this is used in a collection of selected items, see the Search component.
+ * This component can be used to show an item styled as a "tag", optionally with an `X` + "remove"
+ * or with a popover that is shown on click.
  *
  * @return { object } -
  */
-const Tag = ( { id, instanceId, label, remove, removeLabel, screenReaderLabel, className } ) => {
-	screenReaderLabel = screenReaderLabel || label;
-	if ( ! label ) {
-		// A null label probably means something went wrong
-		// @todo Maybe this should be a loading indicator?
-		return null;
-	}
-	const classes = classnames( 'woocommerce-tag', className, {
-		'has-remove': !! remove,
-	} );
-	const labelId = `woocommerce-tag__label-${ instanceId }`;
-	return (
-		<span className={ classes }>
-			<span className="woocommerce-tag__text" id={ labelId }>
+const Tag = withState( {
+	isVisible: false,
+} )(
+	( {
+		id,
+		instanceId,
+		isVisible,
+		label,
+		popoverContents,
+		remove,
+		removeLabel,
+		screenReaderLabel,
+		setState,
+		className,
+	} ) => {
+		screenReaderLabel = screenReaderLabel || label;
+		if ( ! label ) {
+			// A null label probably means something went wrong
+			// @todo Maybe this should be a loading indicator?
+			return null;
+		}
+		const classes = classnames( 'woocommerce-tag', className, {
+			'has-remove': !! remove,
+		} );
+		const labelId = `woocommerce-tag__label-${ instanceId }`;
+		const labelTextNode = (
+			<Fragment>
 				<span className="screen-reader-text">{ screenReaderLabel }</span>
 				<span aria-hidden="true">{ label }</span>
-			</span>
+			</Fragment>
+		);
 
-			{ remove && (
-				<IconButton
-					className="woocommerce-tag__remove"
-					icon={ <Gridicon icon="cross-small" size={ 24 } /> }
-					onClick={ remove( id ) }
-					label={ removeLabel }
-					aria-describedby={ labelId }
-				/>
-			) }
-		</span>
-	);
-};
+		return (
+			<span className={ classes }>
+				{ popoverContents ? (
+					<Button
+						className="woocommerce-tag__text"
+						id={ labelId }
+						onClick={ () => setState( () => ( { isVisible: true } ) ) }
+						isToggled={ isVisible }
+					>
+						{ labelTextNode }
+					</Button>
+				) : (
+					<span className="woocommerce-tag__text" id={ labelId }>
+						{ labelTextNode }
+					</span>
+				) }
+				{ popoverContents &&
+					isVisible && (
+						<Popover onClose={ () => setState( () => ( { isVisible: false } ) ) }>
+							{ popoverContents }
+						</Popover>
+					) }
+				{ remove && (
+					<IconButton
+						className="woocommerce-tag__remove"
+						icon={ <Dashicon icon="dismiss" size={ 20 } /> }
+						onClick={ remove( id ) }
+						label={ removeLabel }
+						aria-describedby={ labelId }
+					/>
+				) }
+			</span>
+		);
+	}
+);
 
 Tag.propTypes = {
+	/**
+	 * The name for this item, displayed as the tag's text.
+	 */
+	label: PropTypes.string.isRequired,
```

Comment:
Usually proptypes should be alphabetical, but I'm not sure that every component follows this.

Reply:
Is that also the case for required props? I thought required props had to be on top of the optional ones.

Reply:
I'm pretty sure we've never separated them out like that intentionally

Reply:
Oh, ok. I'm not sure why I was convinced I'd read somewhere we were planning to do it. I just reverted it. :+1: